### ISSUE TITLE
PyCharm Professional 2020.2 compatiblity

### ISSUE THREAD

**Issue by andormarkus** [JetBrains/embeddedProjectJdk#8]
Hello,

Based on compatibiliy [page](https://plugins.jetbrains.com/plugin/10480-embeddedprojectjdk/versions)  it is not compatible with the latest PyCharm versions.
Please can you update, thus it will be compatible?

Thanks,
Andor

**Comment by yole** [comment]
What do you intend to use the plugin for in PyCharm?

**Comment by andormarkus** [comment]
Hi @yole 

Short answer: The same purpose what is described in the plugins Overview section
Why: PyCharm store the SSH interpreter related information`jdk.table.xml` file

Long answer:
90% of our work is against a Spark (pypspark) cluster, and we are using the SSH interpreter to execute the code on the cluster. PyCharm stores the SSH interpreter related information`jdk.table.xml` in the `JetBrains/PyCharm####.#/options/` directory.

We want to achieve two things: 
- make the `jdk.table.xml` part of the project not part of the IDE settings like we can achieve with `sshConfigs.xm` or `deployment.xml`. If you open the project only the project related SSH interpreter will be available for you
- If it is part of the project than it can be shared across the developers

We tried to use IDE settings through a settings repository﻿ however the `jdk.table.xml` will not be uploaded to repo. Second issue with this the scope of the `jdk.table.xml`global not project focused.

Thanks,
Andor

**Comment by edisongustavo** [comment]
I'm also interested in having PyCharm Pro 2020.2 compatibility.

My use case is that we have a monorepo with multiple projects inside:

```
.
├── project-1
│   ├── .idea
│   ├── Makefile
│   ├── pyproject.toml
│   ├── README.md
│   ├── requirements-dev.in
│   ├── requirements-dev.txt
│   ├── requirements.in
│   ├── requirements.txt
│   ├── setup.py
│   ├── src
│   └── test
├── project-2
│   ├── .idea
│   ├── Makefile
│   ├── pyproject.toml
│   ├── README.md
│   ├── requirements-dev.in
│   ├── requirements-dev.txt
│   ├── requirements.in
│   ├── requirements.txt
│   ├── setup.py
│   ├── src
│   └── test
...
```

We use conda to create the virtual environment for each of these projects, and then I configure the Python interpreter for each of these projects. So my `jdk.table.xml` looks like this:

```
<application>
  <component name="ProjectJdkTable">
    <jdk version="2">
      <name value="Python 3.7 (project-1)" />
      <type value="Python SDK" />
      <version value="Python 3.7.4" />
      <homePath value="$USER_HOME$/miniconda3/envs/project1/bin/python" />
      <roots>
        <classPath>
          <root type="composite">
            <root url="file://$USER_HOME$/miniconda3/envs/project-1/lib/python3.7" type="simple" />
            <root url="file://$USER_HOME$/miniconda3/envs/project-1/lib/python3.7/lib-dynload" type="simple" />
            <root url="file://$USER_HOME$/miniconda3/envs/project-1/lib/python3.7/site-packages" type="simple" />
            <root url="file://$USER_HOME$/.cache/JetBrains/PyCharm2020.2/python_stubs/-704480173" type="simple" />
            <root url="file://$APPLICATION_HOME_DIR$/plugins/python/helpers/python-skeletons" type="simple" />
            <root url="file://$APPLICATION_HOME_DIR$/plugins/python/helpers/typeshed/stdlib/3.7" type="simple" />
            <root url="file://$APPLICATION_HOME_DIR$/plugins/python/helpers/typeshed/stdlib/3" type="simple" />
            <root url="file://$APPLICATION_HOME_DIR$/plugins/python/helpers/typeshed/stdlib/2and3" type="simple" />
            <root url="file://$APPLICATION_HOME_DIR$/plugins/python/helpers/typeshed/third_party/3" type="simple" />
            <root url="file://$APPLICATION_HOME_DIR$/plugins/python/helpers/typeshed/third_party/2and3" type="simple" />
          </root>
        </classPath>
        <sourcePath>
          <root type="composite" />
        </sourcePath>
      </roots>
      <additional ASSOCIATED_PROJECT_PATH="$USER_HOME$/workspace/root-of-gitrepo" />
    </jdk>
    ...
```

It's ok for us to force everyone to have the git repo in `$USER_HOME$/workspace/root-of-gitrepo` and the conda installation in `$USER_HOME$/miniconda3/`, so we should be able to add `jdk.table.xml` to VCS.

**Comment by hb2638** [comment]
I create a new directory per python project and each project has it's own venv. I have a script that bootstraps pycharm when I create  a new project (It creates the XML files so pycharm knows which venv to use as well as give it an appropriate name and defaults several environment variables and configures the templates for pytest/unitests and python).

**Comment by mfilippov** [comment]
Fixed in 2.5
